File: app/src/main/java/dev/suhockii/lifetest/di/module/LocalDataModule.java
package dev.suhockii.lifetest.di.module;

import androidx.room.Room;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dev.suhockii.lifetest.data.local.ProductsDb;
import dev.suhockii.lifetest.data.local.dao.ProductDao;
import dev.suhockii.lifetest.data.local.dao.ProductDetailsDao;

@Module
public class LocalDataModule {
    @Provides
    @Singleton
    ProductsDb provideProductsDb(Context appContext) {
        return Room.databaseBuilder(appContext, ProductsDb.class, "products.db").build();
    }

    @Singleton
    @Provides
    ProductDao provideProductDao(ProductsDb db) {
        return db.getProductDao();
    }

    @Singleton
    @Provides
    ProductDetailsDao provideProductDetailsDao(ProductsDb db) {
        return db.getProductDetailsDao();
    }
}

File: app/src/main/java/dev/suhockii/lifetest/di/module/LocalDataModule.kt
package dev.suhockii.lifetest.di.module

import androidx.room.Room
import android.content.Context

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import dev.suhockii.lifetest.data.local.ProductsDb
import dev.suhockii.lifetest.data.local.dao.ProductDao
import dev.suhockii.lifetest.data.local.dao.ProductDetailsDao

@Module
class LocalDataModule {
    @Provides
    @Singleton
    fun provideProductsDb(appContext: Context): ProductsDb {
        return Room.databaseBuilder(appContext, ProductsDb::class.java, "products.db").build()
    }

    @Singleton
    @Provides
    fun provideProductDao(db: ProductsDb): ProductDao {
        return db.productDao
    }

    @Singleton
    @Provides
    fun provideProductDetailsDao(db: ProductsDb): ProductDetailsDao {
        return db.productDetailsDao
    }
}
